### ISSUE TITLE
False positive on super(cls, self) when self is nonlocal

### ISSUE THREAD

**Issue by crusaderky** [python/mypy#7747]
Python 3.7.4, mypy 0.740

If a function locally defined inside a bound method invokes super(cls, self) and self is a nonlocal variable, mypy outputs a false positive.

Workarounds:
- ``# type: ignore``
- pass self to the local function, so that self becomes local
- use a lambda

```python
class C:
    def f(self) -> int:
        return 1


class D(C):
    def f(self) -> int:
        def f1() -> int:
            # mypy output:
            # error: super() requires one or more positional arguments in enclosing function
            return super(D, self).f() + 2 

        def f2(self: D) -> int:
            return super().f() + 3

        f3 = lambda: super(D, self).f() + 4

        return f1() + f2(self) + f3()


print(D().f())  # 12

**Comment by ilevkivskyi** [comment]
I cannot reproduce this on master, I assume this was fixed. Please feel free to reopen if you will find a repro that triggers this on master.

**Comment by crusaderky** [comment]
Confirmed, cannot reproduce with git tip
